src/main/java/Duke.java

```diff
@@ -1,10 +1,111 @@
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+
+import java.util.Scanner;
+
 public class Duke {
+    private static final String EVENT = "event";
+    private static final String TODO = "todo";
+    private static final String DEADLINE = "deadline";
+
+    private static final String SEPARATOR = "____________________________________________________________";
+    private static final int LENGTH = 100;
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        welcomeMsg();
+
+        Scanner input = new Scanner(System.in);
+        String val = input.nextLine();
+
+        Task[] tasks = new Task[LENGTH];
+        int length = 0;
+
+        while(!val.equals("bye")){
+            System.out.println(SEPARATOR);
+            if(val.equals("list")){
+                printList(tasks, length);
+            } else if (val.contains("unmark")) {
+                markTask(val, tasks, false, "OK, I've marked this task as not done yet:");
+            } else if (val.contains("mark")){
+                markTask(val, tasks, true, "Nice! I've marked this task as done:");
+            } else{
+                length += addTask(val, tasks, length);
+            }
+            System.out.println(SEPARATOR);
+            val = input.nextLine();
+        }
+
+        byeMsg();
+    }
+
+    private static int addTask(String val, Task[] tasks, int length) {
+        if(val.contains(TODO)){
+            if(val.substring(TODO.length()).trim().isEmpty()){
+                emptyError(TODO);
+                return 0;
+            }
+
+            tasks[length] = new Todo(val.substring(TODO.length()));
+
+        } else if (val.contains(DEADLINE)) {
+            if(val.substring(DEADLINE.length()).trim().isEmpty()){
+                emptyError(DEADLINE);
+                return 0;
+            }
+            tasks[length] = new Deadline(val.substring(DEADLINE.length(), val.indexOf("/")), val.substring((val.indexOf("/") + 4)));
+
+        } else if (val.contains(EVENT)) {
+            if(val.substring(EVENT.length()).trim().isEmpty()){
+                emptyError(EVENT);
+                return 0;
+            }
+            tasks[length] = new Event(val.substring(EVENT.length(), val.indexOf("/")), val.substring((val.indexOf("/") + 4)));
+
+        }else{
+            System.out.println("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+            return 0;
+        }
+        length++;
+        System.out.println("Now you have " + length + " tasks in the list.");
+        return 1;
+    }
+
+    private static void emptyError(String taskName){
```

Comment:
Method names should be verbs as per the Java coding standard. Also good to indicate what exactly is empty and making it clear to infer from the method name. Consider renaming it to something like `showEmptyDescriptionError()`